lib/protobuf/field/enum_field.rb

```diff
@@ -32,30 +32,18 @@ def enum?
         true
       end
 
+      def coerce!(value)
+        enum_value = type_class.fetch(value)
+        fail TypeError, "Invalid Enum value: #{value.inspect} for #{name}" unless enum_value
```

Comment:
No this is an enum, and fetch is the API to get th enum object from the tag or bame